pkg/sinks/forward/forward.go

```diff
@@ -524,10 +545,14 @@ func (df *DataForward) updateSinkWriteMetrics(writeCount int, writeBytes float64
 	metrics.WriteMessagesCount.With(labels).Add(float64(writeCount))
 	metrics.WriteBytesCount.With(labels).Add(writeBytes)
 
+	// Add write processing time metric
+	metrics.WriteProcessingTime.With(labels).Observe(float64(time.Since(writeStart).Microseconds()))
+
 	// if this is for Fallback Sink, increment specific metrics as well
 	if isFallback {
 		metrics.FbSinkWriteMessagesCount.With(labels).Add(float64(writeCount))
 		metrics.FbSinkWriteBytesCount.With(labels).Add(writeBytes)
+		metrics.FbSinkWriteProcessingTime.With(labels).Observe(float64(time.Since(writeStart).Microseconds()))
```

Comment:
if it's fallback sink, do we publish for both WriteProcessingTime and FbSinkWriteProcessingTime?

Reply:
Yes, for fallback sink, normal sink metrics are being published in addition to fb sink.
For eg: if we are writing to fb sink, then `WriteMessagesCount` is incremented (successfully written to fb sink after try with normal sink), and we increment `FbSinkWriteMessagesCount` as well to know the message count written specifically to fb sink.

Reply:
I see, so for a batch size of 100, if I successfully wrote 80 to `primary-sink` and 20 to `fallback` sink, the metric will show `WriteMessageCount` as 100 and and `FbSinkWriteMessagesCount` 20. The label is the vertex name so it makes sense. Thanks.